### PR TITLE
BalancePrecompiled add listCaller interface

### PR DESCRIPTION
**Pull request by wenlinlee**

### CODE REVIEW

File: bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp [hunk 1]
```diff
@@ -54,6 +55,8 @@ BalancePrecompiled::BalancePrecompiled(crypto::Hash::Ptr _hashImpl) : Precompile
         getFuncSelector(BALANCE_METHOD_REGISTER_CALLER, _hashImpl);
     name2Selector[BALANCE_METHOD_UNREGISTER_CALLER] =
         getFuncSelector(BALANCE_METHOD_UNREGISTER_CALLER, _hashImpl);
+    name2Selector[BALANCE_METHOD_LIST_CALLER] =
```
Comment (bxq2011hust) [inline bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp:58]: add ut for the method

File: bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp [hunk 2]
```diff
@@ -527,3 +522,50 @@ void BalancePrecompiled::unregisterCaller(
...
+    auto table = _executive->storage().openTable(SYS_BALANCE_CALLER);
+    if (!table)
+    {
+        PRECOMPILED_LOG(INFO) << BLOCK_NUMBER(blockContext.number())
+                              << LOG_BADGE("BalancePrecompiled")
+                              << LOG_DESC("listCaller failed, caller table not exist");
+        BOOST_THROW_EXCEPTION(protocol::PrecompiledError("caller table not exist."));
+    }
+
+    auto keyCondition = std::make_shared<storage::Condition>();
```
Comment (kyonRay) [inline bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp:543]: Unnecessary make_shared

File: bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp [hunk 3]
```diff
@@ -527,3 +522,50 @@ void BalancePrecompiled::unregisterCaller(
...
+                              << LOG_DESC("listCaller failed, caller table is empty");
+        BOOST_THROW_EXCEPTION(protocol::PrecompiledError("caller table is empty."));
+    }
+    Addresses addresses;
+    for (auto& it : tableKeyList)
+    {
+        auto entry = table->getRow(it);
+        if (entry && entry->get() == "1")
+        {
+            it = "0x" + it;
```
Comment (kyonRay) [inline bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp:559]: It is not recommended to modify iterate values in this way.

**Review by kyonRay — APPROVED** [review]

File: bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp (hunk 1 above)
Comment (wenlinlee) [reply bcos-executor/src/precompiled/extension/BalancePrecompiled.cpp:58]: ok，next week add balancePrecompiled ut

**Review by LucasLi1024 — APPROVED** [review]

**Pull request merged**